### PR TITLE
use pydantic settings

### PR DESCRIPTION
**Pull request by vincentsarago**
proposed changed for #639 

This PR is moving the env configuration into `ApiSettings`

### CODE REVIEW

**Review by jonhealy1 — APPROVED** [review]
LGTM

File: stac_fastapi/types/stac_fastapi/types/config.py [hunk 1]
```diff
@@ -22,6 +22,11 @@ class ApiSettings(BaseSettings):
     # `pydantic.BaseSettings` instead
     default_includes: Optional[Set[str]] = None
 
+    stac_fastapi_title: str = "stac-fastapi"
+    stac_fastapi_description: str = "stac-fastapi"
+    stac_fastapi_version: str = "0.1"
+    stac_fastapi_landing_id: str = "stac-fastapi"
```
Comment (vincentsarago) [inline stac_fastapi/types/stac_fastapi/types/config.py:28]: as discussed in #639 I think it makes more sense for this variable to be named `landing_id`

Comment (vincentsarago) [inline stac_fastapi/types/stac_fastapi/types/config.py:28]: I've also reverted to `stac-fastapi` default as it was before 

https://github.com/stac-utils/stac-fastapi/blob/315cfaefef2ab2773ae087228439ba0371fb7372/stac_fastapi/types/stac_fastapi/types/core.py#L258

**Comment by vincentsarago** [comment]
#657 @jonhealy1 could you check with the latest change 🙏 

@m-mohr I'll wait for your 👍

File: stac_fastapi/types/stac_fastapi/types/config.py (hunk 1 above)
Comment (jonhealy1) [reply stac_fastapi/types/stac_fastapi/types/config.py:28]: Looks good to me. The version is the version number of the User's stac_fastapi, not this version ie. soon to be 2.5.0

**Pull request merged**